File: Assignment21/Program5.c
// Time Complexity : N

#include<stdio.h>
#include<stdlib.h>

void DigitsSum(int Arr[], int iLength)
{
    int iCnt = 0;
    int iDigit = 0;

    printf("Summation of Digits of Numbers are : \n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        int iSum = 0;
        int iTemp = Arr[iCnt];

        while(iTemp != 0)
        {
            iDigit = iTemp % 10;
            iSum = iSum + iDigit;
            iTemp = iTemp / 10;
        }

        printf("%d\n",iSum);
    }
}

int main ()
{
    int iSize = 0, iRet = 0, iCnt = 0;
    int *p = NULL;

    printf("Enter number of elements : ");
    scanf("%d",&iSize);

    p = (int*)malloc(iSize * sizeof(int));

    if(p == NULL)
    {
        printf("Unable to allocate memory\n");
        return -1;
    }

    printf("Enter elements : \n");

    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        scanf("%d",&p[iCnt]);
    }

    DigitsSum(p, iSize);

    free(p);

    return 0;
}

File: Assignment21/Program5.java
class Logic
{
    void printDivisibleBy2and3(int iNum)
    {
        int iCnt = 0;

        System.out.println("The numbers divisible by 2 and 3 are : ");
        
        for(iCnt = 1; iCnt <= iNum; iCnt++)
        {
            if(iCnt % 2 == 0 && iCnt % 3 == 0)
            {
                System.out.println(iCnt);       
            }
        }

        System.out.println(iCnt);
    }
}

class Program5
{
    public static void main(String A[])
    {
        Logic lobj = new Logic();
        lobj.printDivisibleBy2and3(30);  
    }
}